src/FieldWarning/Assets/Units/ShellBehaviour.cs

```diff
@@ -118,33 +126,61 @@ private void Update()
             Vector3 worldForward = transform.TransformDirection(Vector3.forward);
             worldForward = new Vector3(worldForward.x, 0, worldForward.z);
             Vector3 translation = _forwardSpeed * worldForward * Time.deltaTime
-                                  + _verticalSpeed * Vector3.up * Time.deltaTime;
-            transform.LookAt(transform.position + translation);
+                                  + _verticalSpeed * Vector3.up * Time.deltaTime
+                                  + _targetVelocity * Time.deltaTime; /// * Constants.MAP_SCALE;
+            ///transform.LookAt(transform.position + translation);
             transform.Translate(
                     translation,
                     Space.World);
 
             _verticalSpeed -= GRAVITY * Time.deltaTime;
 
-
-            // small trick to detect if shell has reached the target
-            float distanceToTarget = Vector3.Distance(transform.position, _targetCoordinates);
-            if (distanceToTarget > _prevDistanceToTarget)
+            if (_justLaunched)
             {
-                transform.position = _targetCoordinates;
-                Explode();
+                print("instant collision not avoided");
             }
-            _prevDistanceToTarget = distanceToTarget;
-
-            //transform.rotation.SetLookRotation(translation);
+            _justLaunched = false;
         }
 
         private void OnTriggerEnter(Collider other)
         {
-            if (other.GetComponent<CaptureZone>() == null)
+            if (_justLaunched)
             {
-                Explode();
+                _launchPlatform = other;
+                _justLaunched = false;
+                print("instant collision avoided now");
+            }
+            else if(_launchPlatform != other)
+            {
+                UnitDispatcher target = other.gameObject.GetComponentInParent<UnitDispatcher>();
+
+                if (target != null && !_ammo.IsAoe)
+                {
+                    Kinetic(target);
+                }
+                else if (other.GetComponent<CaptureZone>() == null && _ammo.IsAoe)
```

Comment:
The capturezone check should be at the top level, not just for this one case

Reply:
What is the capturezone check actually, or supposed to do? I just tried to keep it how I thought it was when I started working with it, but I really don't know.

Reply:
Capture zones have colliders, so to prevent shells exploding from touching a capture zone, we have to ignore events caused by 'collision' with a zone.

Reply:
We will probably at some point rewrite the zones to not have colliders, and to detect units that enter them in some other way instead, but for now it's a necessary hack and so the shell needs to always ignore collisions with zones.

Reply:
I understand, thank you